feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/sendrequest/SendRequestFragment.java

```diff
@@ -1,33 +1,169 @@
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
 import androidx.lifecycle.Observer;
 import androidx.lifecycle.ViewModelProviders;
 
+import com.example.feelslikemonday.DAO.FollowRequestCallback;
+import com.example.feelslikemonday.DAO.FollowRequestDAO;
+import com.example.feelslikemonday.DAO.VoidCallback;
 import com.example.feelslikemonday.R;
+import com.example.feelslikemonday.model.FollowRequest;
 import com.example.feelslikemonday.ui.followrequest.FollowerRequestViewModel;
+import com.google.android.gms.tasks.OnSuccessListener;
+
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import static androidx.constraintlayout.widget.Constraints.TAG;
 
 public class SendRequestFragment extends Fragment {
 
+    private Button resetButton;
+    private Button sendButton;
+    private EditText usernameEditText;
+    private String recipientUsername;
+    private List<String> requesterUsernames;
+    private FollowRequest followRequest_ok;
+
     private SendRequestViewModel sendRequestViewModel;
+    private static FollowRequestDAO DAO; // = FollowRequestDAO.getInstance();
+
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 
-    public View onCreateView(@NonNull LayoutInflater inflater,
-                             ViewGroup container, Bundle savedInstanceState) {
         sendRequestViewModel = ViewModelProviders.of(this).get(SendRequestViewModel.class);
         View root = inflater.inflate(R.layout.fragment_send_request, container, false);
+
+        resetButton =  root.findViewById(R.id.send_request_reset);
+        sendButton =  root .findViewById(R.id.send_request_send);
+        usernameEditText =  root .findViewById(R.id.send_request_username);
+
         sendRequestViewModel.getText().observe(getViewLifecycleOwner(), new Observer<String>() {
             @Override
             public void onChanged(@Nullable String s) {
                 //do stuff here
             }
         });
+
+        resetButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                usernameEditText.setText("");
+            }
+        });
+
+        usernameEditText.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+            }
+        });
+
+        //https://developer.android.com/guide/topics/ui/notifiers/toasts.html
+        sendButton.setOnClickListener(new View.OnClickListener() {
+            //TODO: pass filename(requesterUsername) variable from MainActivity
+
+            @Override
+            public void onClick(View view) {
+
+                final CountDownLatch signal = new CountDownLatch(1);
```

Comment:
We do not want to use the signal here, it should only be used for testing (it might not even be suitable for that). using this will block our code up and will diminish the efficiency of our callback system